futures/src/channel.rs

```diff
@@ -146,7 +146,7 @@ pub struct ChannelFlowOptions {
 
 impl<T: AsyncRead+AsyncWrite+Send+'static> Channel<T> {
     /// create a channel
-    pub fn create(transport: Arc<Mutex<AMQPTransport<T>>>) -> Box<Future<Item = Self, Error = io::Error> + Send + 'static> {
+    pub fn create(transport: Arc<Mutex<AMQPTransport<T>>>) -> impl Future<Item = Self, Error = io::Error> + Send {
```

Comment:
Why are you removing all the 'static lifetimes?

Reply:
but they are not needed with `impl Trait` anymore

so I just don't see why leave unnecessary restrictions

Reply:
They can be needed in some scenarii actually from what I understand, and it doesn't harm to keep those around I'd say.
If we can add some "guarantees" to the return type "for free", I don't see why we should remove those

Reply:
Are you sure about these "additional guarantees for return type"? 

I believe that `'static` within type like `Box<SomeTrait + 'static>` means just "there is no non-static references inside my trait object", isn't it?

And in case of `impl Trait` compiler already has a full type in its hands so it can easily examine it itself.

Reply:
I'm not 100% sure, but I really see no harm in keeping those markers in place

Reply:
I don't know if there is any harm in them, but they are really not needed :) Even if a generic argument requires `'static` the compiler could check it itself (proof: https://play.rust-lang.org/?gist=037be4712b2acfc8411b2725e7ae94ae&version=stable&mode=debug). Probably these markers are added automatically but I'm not sure.

It just works in a different way than in trait objects.

Reply:
Right

Reply:
I just realized we can remove `Send` marker as well! Should I do it?